osf/migrations/0142_add_spam_fields_to_user.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2018-10-24 17:50
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import osf.models.spam
+import osf.utils.datetime_aware_jsonfield
+import osf.utils.fields
+
+
+TAG_MAP = {
+    'spam_flagged': osf.models.spam.SpamStatus.FLAGGED,
+    'spam_confirmed': osf.models.spam.SpamStatus.SPAM,
+    'ham_confirmed': osf.models.spam.SpamStatus.HAM
+}
+
+def add_spam_status_to_tagged_users(state, schema):
+    OSFUser = state.get_model('osf', 'osfuser')
+    users_with_tag = OSFUser.objects.filter(tags__name__in=TAG_MAP.keys()).prefetch_related('tags')
+    for user in users_with_tag:
+        for tag, value in TAG_MAP.items():
+            if user.tags.filter(name=tag).exists():
```

Comment:
A few things about the data migration:

1. I believe this needs to be changed to user.system_tags.filter(...) instead of user.tags.  tags does not include system tags by default.  (But for some reason `users_with_tags` query above returns users with system tags, so I don't understand yet how that line is working.)

2.  It seems unlikely, but is it possible for a user to have multiple conflicting system tags?  i.e. both `spam_flagged` and `spam_confirmed`.   Since it looks like we were just adding/removing these system tags as the user's spam status was changed, it seems possible during a save a tag could have been added, and the previous tag not removed.  Maybe worth doing a sanity check on staging and eventually prod data to make sure this isn't the case?

3. I don't know the order of magnitude of users that would have one of these spam system tags. I'd think it would be low so we _probably_ wouldn't have to optimize too much.  `bulk_update` is an option however, to avoid all these individual save calls.

4.   It looks like you're intentionally not removing the spam-related system tags.  It's nice to have a backup for immediately after the migration but long-term that backup isn't so useful.   My recommendation would be a follow-up ticket after this ticket is out to remove those tags at a later date.

Reply:
@pattisdr Follow up PR to remove spam system tags makes a lot of sense! Do you think it should be a follow up PR attached to this ticket or a new ticket we can put into a sprint after this is merged?

Reply:
It can be a new ticket we put into the sprint - not a huge rush to remove these system tags, but not a lot of value in keeping them forever, either.